src/util/type.js

```diff
@@ -9,4 +9,14 @@ export function isDOMNode(o){
     typeof Node === "object" ? o instanceof Node : 
     o && typeof o === "object" && typeof o.nodeType === "number" && typeof o.nodeName==="string"
   );
+}
+
+// If constructor is not in object's prototype chain,
+// return object wrapped in constructor
+export function wrapIfNotInstance(object, constructor) {
+  if (!(object instanceof constructor)) {
```

Comment:
You unwittingly reinvented a common pattern I've seen used. I'm not sure what it is called... I'll look it up later.